packages/itwin/tree-widget/src/components/trees/VisibilityTreeEventHandler.ts

```diff
@@ -129,7 +137,16 @@ export class VisibilityTreeEventHandler extends UnifiedSelectionTreeEventHandler
       .pipe(
         mergeMap((changes) => this.changeVisibility(changes)),
       )
-      .subscribe();
+      .subscribe({
+        complete: () => {
+          this._isChangingVisibility = false;
+          void this.updateCheckboxes();
```

Comment:
Looks like this function can be extracted into local variable to avoid duplicating it for `complete` and `error` callbacks